content/articles/2017-11-01-lubridate-1.7.0.Rmd

```diff
@@ -0,0 +1,135 @@
+  author: Heather Zabriskie
+categories: [package]
+---
+
+```{r setup, include = FALSE}
+library("lubridate")
+knitr::opts_chunk$set(collapse = TRUE, comment = "##=>")
+```
+
+We are pleased to announce that `lubridate` 1.7.0 is now on CRAN! For a complete
+set of changes please see the release
+[NEWS](https://github.com/tidyverse/lubridate/releases/tag/v1.7.0). Here is a
+brief walk-through of the prominent new features.
+
+## Built-in CCTZ and much faster `update` and `force_tz`
+
+From this version Lubridate relies on google's
+[CCTZ](https://github.com/google/cctz) library for the date-time updates and
+time-zone manipulation. This means that `force_tz`, `update`, `round`ing and a
+range of arithmetic operations on timespans are now considerably faster.
+
+## Vectorized time-zone manipulation
+
+Often date-time data comes with heterogeneous time-zones. For example, you might
+have a character vector of date-times and a separate vector of time-zones. In
+such cases you can parse your date-time vector in one time zone (`UTC`) and then
+"enforce" the heterogeneous time-zones with `force_tzs`.
+
+
+```{r}
+x <- ymd_hms(c("2009-08-01 00:00:00", "2009-08-01 00:00:00"))
+tzs <- c("America/New_York", "Europe/Amsterdam")
+
+force_tzs(x, tzones = tzs)
+
+force_tzs(x, tzones = tzs, tzone_out = "America/New_York")
+```
+
+Note that first `force_tzs` call produced a vector of instants in UTC time-zone,
+for the second call we specified the desired time-zone of the output vector with
+the `tzone_out` argument. This is needed as R's date-time vectors cannot
+represent heterogeneous time-zones. For the same reason, the counterpart of
+`with_tz`, `with_tzs`, does not exist. Instead, `local_time` should cover the
+`with_tzs` use case in most situations.
+
+New `local_time` function retrieves day clock time of the input vector in the
+specified time zones. It returns a `difftime` object and it is vectorized over
+both date-time and time-zone arguments:
+
+```{r}
+x <- ymd_hms(c("2009-08-01 01:02:03", "2009-08-01 10:20:30"))
+local_time(x, units = "hours")
+
+x <- ymd_hms(c("2009-08-01 00:00:00", "2009-08-01 00:00:00"))
+tzs <- c("America/New_York", "Europe/Amsterdam")
+local_time(x, tzs, units = "hours")
+```
+
+## New `cutoff_2000` parameter in parsing functions
+
+Lower level lubridate parser functions (`parse_datetime2`,`fast_strptime`) now
+accept `cutoff_2000` parameter to determine when the parsing of `yy` format
+should output 19th or 20th century.
+
+```{r}
+x <- c("50-01-01", "70-01-01")
+parse_date_time2(x, "ymd")
+parse_date_time2(x, "ymd", cutoff_2000 = 30)
+```
+
+By default `cutoff_2000` is 68 to comply with R's `strptime` function.
+
+This feature was not propagated to higher level parsing function because base
+`strptime`, on which those occasionally rely, doesn't support this option.
+
+
+## Functions `wday` and `month` are now localized
+
+When `label=TRUE` `wday` and `month` extractors return localized strings:
```

Comment:
This is unfortunately going in the opposite direction to the rest of the tidyverse - I think this is dangerous because it makes it easy to write code that works for you, but not in a different locale.

Reply:
I agree. I am avoiding localization as much as I can but in this case it's pretty harmless  as human readable output should never be used programaticaly, only for printing and reporting. Programs should use numeric output which is the default.  

I actually was planing to add a comment on this in the blog, but removed it eventually.

Reply:
Yeah, I think you should add a comment here - i.e. you should only use these for labels, not for comparisons, otherwise your code might not work on someone else's computer.

Reply:
Ok, fixed.